### PR TITLE
imx6ull-flash: rewrite initialization 

### PR DESCRIPTION
**Pull request by nalajcie**
## Description
- rewrite options parsing
- parse/validate options first, then create necessary partitions
- common logging
- introduce partition names with symlink creation
- select rootfs to be mounted based on SRC secondary boot 

## Motivation and Context
JIRA: DTR-181

Needed for FW upgrade (the same kernel image for primary/secondary boot, current rootfs detection, etc.)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change: **parameter syntax for `imx6ull-flash` changed, PRs for `psu`/`project` would follow**

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: `imx6ull`

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [x] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: storage/imx6ull-flash/flashsrv.c [hunk 1]
```diff
@@ -767,20 +782,138 @@ static void daemonize(void)
...
+				}
+
+				break;
+
+			case 'p': /* start:size[:name] */
+				/* TODO: switch to positional arguments to highlight partition order matter */
+
+				errno = 0;
+				part_start = strtol(optarg, &p, 10);
+				if (*p++ != ':') {
```
Comment (agkaminski) [inline storage/imx6ull-flash/flashsrv.c:817]: `char *p` might be uninitialized in this case

File: storage/imx6ull-flash/flashsrv.c [hunk 2]
```diff
@@ -743,21 +753,26 @@ static void flashsrv_devThread(void *arg)
...
+	int ret;
 
 	p = malloc(sizeof(*p));
 
+	if (p == NULL)
+		return -1;
+
 	p->start = start;
 	p->size = size;
+	p->name = name;
```
Comment (agkaminski) [inline storage/imx6ull-flash/flashsrv.c:768]: `strdup()`? You pass to this function `name` placed in non static storage

Comment (nalajcie) [reply storage/imx6ull-flash/flashsrv.c:768]: I believe `**argv` is mine to rule inside the program? I can strdup if that would be more readable.

Second thought: if we would ever add partition table support, `strdup` hete might be more straightforward, will fix in a sec

File: storage/imx6ull-flash/flashsrv.c (hunk 1 above)
Comment (nalajcie) [reply storage/imx6ull-flash/flashsrv.c:817]: how? if the entire input string is valid, `strtol` will set `*endptr` as a pointer to `\0`

Comment (agkaminski) [reply storage/imx6ull-flash/flashsrv.c:817]: Yes, didn't notice p in strtol

**Review by agkaminski — APPROVED** [review]

**Pull request merged**